drivers/saul/init_devs/auto_init_saul_nrf_vddh.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     sys_auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of internal voltage sensor directly mapped to SAUL reg
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "log.h"
+#include "saul_reg.h"
+#include "saul/periph.h"
+#include "periph/adc.h"
+
+static int _read_voltage(const void *dev, phydat_t *res)
+{
+    (void)dev;
+
+    /* GPIO reference voltage / external output supply voltage in high voltage mode */
+    const uint8_t vref_dV[] = { 18, 21, 24, 27, 30, 33, 0, 18 };
+    uint8_t idx = NRF_UICR->REGOUT0 & UICR_REGOUT0_VOUT_Msk;
+
+    int raw = adc_sample(NRF52_VDDHDIV5, ADC_RES_12BIT) * 5 * vref_dV[idx] * 100;
```

Comment:
```suggestion
    const uint8_t vref_deci_volt[] = { 18, 21, 24, 27, 30, 33, 0, 18 };
    uint8_t idx = NRF_UICR->REGOUT0 & UICR_REGOUT0_VOUT_Msk;

    int raw = adc_sample(NRF52_VDDHDIV5, ADC_RES_12BIT) * 5 * vref_deci_volt[idx] * 100;
```

Let's avoid camel case. I also think that deci_volt is easier to understand than dV.